server/public/model/config.go

```diff
@@ -2908,10 +2912,18 @@ func (s *DataRetentionSettings) SetDefaults() {
 		s.MessageRetentionDays = NewInt(DataRetentionSettingsDefaultMessageRetentionDays)
 	}
 
+	if s.MessageRetentionHours == nil {
+		s.MessageRetentionHours = NewInt(DataRetentionSettingsDefaultMessageRetentionHours)
+	}
+
 	if s.FileRetentionDays == nil {
 		s.FileRetentionDays = NewInt(DataRetentionSettingsDefaultFileRetentionDays)
 	}
 
+	if s.FileRetentionHours == nil {
+		s.FileRetentionHours = NewInt(DataRetentionSettingsDefaultFileRetentionHours)
+	}
```

Comment:
Should the code respect the value of `MessageRetentionHours` when setting the default? I.e.g if `MessageRetentionHours` is set, use that setting *24 as hours, otherwise use the default.

Reply:
The default needs to be 0 here otherwise it will lead to hours being used over days by default but we want that to be intentional. We don't know the actual value of `MessageRetentionDays` at this point anyway because we can't read the env var value